wsidicom/instance.py

```diff
@@ -817,23 +961,67 @@ def read_frame(self, frame_index: int) -> bytes:
 
     def _parse_pixel_data(self) -> List[Tuple[int, int]]:
         """Parse file pixel data, reads frame positions.
-        Note that fp needs to be positionend at pixel data.
+        Note that fp needs to be positionend at Extended offset table (EOT)
+        or Pixel data. An EOT can be present before the pixel data, and must
+        then not be empty. A BOT most always be the first item in the Pixel
+        data, but can be empty (zero length). If EOT is used BOT must be empty.
 
         Returns
         ----------
         List[Tuple[int, int]]
             List of frame positions and lenghts
         """
-        frame_positions = self._read_frame_positions()
-        fragment_count = len(frame_positions)
+        tag = self._fp.read_tag()  # Either EOT or Pixel data tag
+        frame_positions = []
+        table = None
+        table_type = 'bot'
+        if tag == Tag('ExtendedOffsetTable'):
+            table_type = 'eot'
+            table = self._read_eot()
+            tag = self._fp.read_tag()
+
+        if tag == Tag('PixelData'):
```

Comment:
This if-tree is a little bit hard to follow, reorder to put special paths first and use binomial variable to document the the if-clauses if needed.